### ISSUE TITLE
Require to show Dimension.Hierarchy name when using execute_mdx_dataframe function

### ISSUE THREAD

**Issue by tatwong** [cubewise-code/tm1py#920]
**Describe what did you try to do with TM1py**

Using this sample:
https://github.com/cubewise-code/tm1py-samples/blob/master/Data/cube_data_into_pandas_dataframe.py

This is a sample MDX:
```
SELECT
  NON EMPTY
   {[M Flight Schedule Detail].[M Flight Schedule Detail].[Arising]}
  ON COLUMNS ,
  NON EMPTY
   {TM1FilterByLevel({Descendants({[Date].[2023-04]})}, 0)}
   * {[Sector].[Sector by Region].[All Sectors].Children}
   * {[Sector].[Sectors by AD].[Departure],[Sector].[Sectors by AD].[Arrival]}
   * {[Sector].[Departure Airport].[All Departure Airports].Children}
   * {[Sector].[Arrival Airport].[All Arrival Airports].Children}
   * {TM1FilterByLevel({TM1SubsetAll([Aircraft])}, 0)}
   * {TM1FilterByLevel({TM1SubsetAll([Flight ID])}, 0)}
   * {TM1FilterByLevel({TM1SubsetAll([Service Type])}, 0)}
ON ROWS
FROM [Flight Schedule Detail]
WHERE
  (
   [Time].[Time].[All Hours]
  )
```

**Describe what's not working the way you expect**

Currently in our example, only the dimension names are shown in the dataframe, but since we are using many hierarchies in the MDX, the column names from the data frame became confusing:

![Ronan Wang | Microsoft Teams 2023-05-23 15-52-41](https://github.com/cubewise-code/tm1py/assets/43463864/c4d29980-37c1-4b29-9554-3e6bfa7c827f)

Is there anyway to display on the columns like [Sector].[Sector by Region], [Sector].[Sectors by AD], etc, instead of just Sector?

**Comment by MariusWirtz** [comment]
Hi @tatwong ,

You are using the `build_pandas_dataframe_from_cellset` function to build the data frame based on a dictionary.
You can use the `execute_mdx_dataframe` function to retrieve a data frame directory. This will be more efficient from a code and resource (Memory, CPU) perspective.

When using the `execute_mdx_dataframe` function the pandas data frame by default omits the hierarchy information in the column headers and displays only the dimension header!
In most situations, this isn't a problem, because the data frame columns are ordered based on the MDX.

In your MDX the selections on the rows you have are:
- `[Date].[Date]`
- `[Sector].[Sector by Region]`
- `[Sector].[Sectors by AD]`
- `[Sector].[Departure Airport]`
- `[Sector].[Arrival Airport]`
- `[Aircraft].[Aircraft]`
- `[Flight ID].[Flight ID]`
- `[Service Type].[Service Type]`

As selections on the columns you have:
- `[M Flight Schedule Detail].[M Flight Schedule Detail]`

Based on those selections the columns in the data frame should be:

- `[Date].[Date]`
- `[Sector].[Sector by Region]`
- `[Sector].[Sectors by AD]`
- `[Sector].[Departure Airport]`
- `[Sector].[Arrival Airport]`
- `[Aircraft].[Aircraft]`
- `[Flight ID].[Flight ID]`
- `[Service Type].[Service Type]`
- `[M Flight Schedule Detail].[M Flight Schedule Detail]`

Can you confirm that the headers in the data frame are deterministic and based on the orders of the selections in the MDX?

**Comment by MariusWirtz** [comment]
@tatwong I hope you can work with the suggested "workaround" to determine the column headers based on the MDX query.

Nonetheless, I agree that the missing hierarchy information is a limitation of the `execute_mdx_dataframe` function. 

I think we should introduce an optional argument to this function so that a user could control if data frame column headers are in the current format of `dimension` or instead `dimension.hierarchy` or `hierarchy`

**Comment by MariusWirtz** [comment]
Resolved with #961
